pkg/services/store/object/object.proto

```diff
@@ -0,0 +1,259 @@
+syntax = "proto3";
+package object;
+
+option go_package = "./;object";
+
+// Will be replaced with something from the SDK
+message UserInfo {
+  // internal grafana user ID
+  int64 id = 1;
+
+  // login name
+  string login = 2; // string ID? 
+}
+
+// The canonical object/document data -- this represents the raw bytes and storage level metadata
+message RawObject {
+  // Unique ID
+  string UID = 1;
+
+  // Identify the object kind.  This kind will be used to apply a schema to the body and
+  // will trigger additional indexing behavior.
+  string kind = 2;
+
+  // Time in epoch milliseconds that the object was modified
+  int64 modified = 3;
+
+  // Who modified the object
+  UserInfo modified_by = 4;
+
+  // Content Length
+  int64 size = 5;
+
+  // MD5 digest of the body
+  string ETag = 6;
+
+  // Raw bytes of the storage object.  The kind will determine what is a valid payload
+  bytes body = 7;
+
+  // The version will change when the object is saved. It is not necessarily sortable
+  //
+  // NOTE: currently managed by the dashboard+dashboard_version tables
+  string version = 8;
+
+  // optional "save" or "commit" message
+  //
+  // NOTE: currently managed by the dashboard_version table, and will be returned from a "history" command
+  string comment = 9;
+
+  // Location (path/repo/etc) that defines the canonocal form
+  //
+  // NOTE: currently managed by the dashboard_provisioning table
+  string sync_src = 10;
+
+  // Time in epoch milliseconds that the object was last synced with an external system (provisioning/git)
+  //
+  // NOTE: currently managed by the dashboard_provisioning table
+  int64 sync_time = 11;
+}
+
+// Report error while working with objects
+// NOTE: real systems at scale will contain errors.
+message ObjectErrorInfo {
+  // Match an error code registry?
+  int64 code = 1;
+
+  // Simple error display
+  string message = 2;
+
+  // Details encoded in JSON
+  bytes details_json = 3;
+}
+
+// Reference to another object outside itself
+// This message is derived from the object body and can be used to search for references.
+// This does not represent a method to declare a reference to another object.
+message ExternalReference {
+  // datasource (instance), dashboard (instance),
+  string kind = 1;
+
+  // prometheus / heatmap, heatamp|prometheus
+  string type = 2;  // flavor
```

Comment:
having both kind and type is going to be confusing :) 

Not great but maybe less confusing 
* objectKind
* subType

Reply:
I'll punt for now... I think this is a much larger discussion/clarification where we need to consolidate language.  We currently use "type" for all the flavors.

Given that ExternalReference is not used by protobuf (it is a property of summary), we can move it out